File: frontend/src/types/enums.ts
/**
 * String-literal mirrors of the backend's Python enumerations
 * (`backend/src/db/enums.py` and `backend/src/simulation/fleet_tracker.py`).
 *
 * Plain `enum` is intentionally avoided (the project's `erasableSyntaxOnly`
 * TypeScript setting forbids constructs that require runtime support) in
 * favor of string literal union types plus a `satisfies`-checked tuple of
 * every value, which is enough to both type-check exhaustively and drive a
 * `<select>` option list without any generated runtime code.
 */

/** Lifecycle status of a `WorkdayPlan`. */
export type WorkdayStatus = 'DRAFT' | 'ACTIVE' | 'COMPLETED'

export const WORKDAY_STATUSES = ['DRAFT', 'ACTIVE', 'COMPLETED'] as const satisfies readonly WorkdayStatus[]

/** Role a `RouteStop` plays within its vehicle's visiting sequence. */
export type RouteStopType = 'DEPOT_START' | 'DEPOT_PICKUP' | 'CUSTOMER_DELIVERY' | 'DEPOT_END'

/** Kind of dynamic disruption recorded by a `SimulationEvent`. */
export type SimulationEventType = 'TRAFFIC_INCIDENT' | 'URGENT_ORDER'

/** Operational status of a vehicle at a given simulated instant (`FleetTracker.VehicleStatus`). */
export type VehicleStatus = 'driving' | 'waiting' | 'serving' | 'on_break' | 'idle_at_depot' | 'finished'

File: backend/src/db/enums.py
"""
Enumerations shared by the persistence layer's ORM models and API schemas.

Defining these enumerations independently of `backend/src/db/models.py`
allows `backend/src/api/schemas/` to import the same vocabulary without
depending on SQLAlchemy, keeping the Pydantic DTOs a pure data-validation
layer.
"""

from __future__ import annotations

import enum


class WorkdayStatus(str, enum.Enum):
    """Lifecycle status of a `WorkdayPlan`."""

    DRAFT = "DRAFT"
    ACTIVE = "ACTIVE"
    COMPLETED = "COMPLETED"


class RouteStopType(str, enum.Enum):
    """Role a `RouteStop` plays within its vehicle's visiting sequence."""

    DEPOT_START = "DEPOT_START"
    DEPOT_PICKUP = "DEPOT_PICKUP"
    CUSTOMER_DELIVERY = "CUSTOMER_DELIVERY"
    DEPOT_END = "DEPOT_END"


class SimulationEventType(str, enum.Enum):
    """Kind of dynamic disruption recorded by a `SimulationEvent`."""

    TRAFFIC_INCIDENT = "TRAFFIC_INCIDENT"
    URGENT_ORDER = "URGENT_ORDER"
